platforms/cpu/src/CpuKernels.cpp

```diff
@@ -166,6 +166,12 @@ class CpuCalcForcesAndEnergyKernel::InitForceTask : public ThreadPool::Task {
                 posq[4*i+1] = (float) posData[i][1];
                 posq[4*i+2] = (float) posData[i][2];
             }
+        
+        // Check for invalid positions.
+        
+        for (int i = 4*start; i < 4*end; i++)
+            if (posq[i] != posq[i])
```

Comment:
-ffast-math implies -ffinite-math-only, which tells the compiler it can assume nothing will ever be nan or infinite.  So yes, that would break this!